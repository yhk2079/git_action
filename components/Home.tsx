import * as React from 'react';
import styled from 'styled-components';

const Title = styled.div`
  display: flex;
  justify-content: center;
  color: ${({ theme }) => theme.colors.primary};
`;

const Home = React.memo(function Home() {
  return (
    <div>
      <Title>potato</Title>
    </div>
  );
});

export default Home;
