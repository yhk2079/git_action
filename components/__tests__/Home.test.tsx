import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../utils/theme';
import MockProvider from '../../utils/MockProvider';
import Home from '../Home';

test('should render Home', async () => {
  const { getByText, asFragment } = render(
    <MockProvider>
      <ThemeProvider theme={theme}>
        <Home />
      </ThemeProvider>
    </MockProvider>,
  );
  expect(getByText('potato')).toBeInTheDocument();

  expect(asFragment()).toMatchSnapshot();
});
