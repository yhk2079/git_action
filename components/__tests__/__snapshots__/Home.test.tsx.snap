// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Home__Title-sc-7m4uo6-0 fpFEhU"
    >
      Hi potato!!!!!
    </div>
  </div>
</DocumentFragment>
`;
